File: app/src/main/java/com/example/deltatask3/database/PokemonTypeConverter.java
package com.example.deltatask3.database;

import androidx.room.TypeConverter;

import com.example.deltatask3.utils.Pokemon;
import com.google.gson.Gson;

public class PokemonTypeConverter {

    @TypeConverter
    public static Pokemon stringToPokemon(String value){
        if (value==null) return new Pokemon("Name");
        return new Gson().fromJson(value,Pokemon.class);
    }

    @TypeConverter
    public static String pokemonToString(Pokemon pokemon){
        return new Gson().toJson(pokemon);
    }
}

File: app/src/main/java/com/example/deltatask3/database/PokemonTypeConverter.kt
package com.example.deltatask3.database

import androidx.room.TypeConverter
import com.example.deltatask3.models.Pokemon
import com.google.gson.Gson

object PokemonTypeConverter {
    @JvmStatic
    @TypeConverter
    fun stringToPokemon(value: String?): Pokemon =
        if (value == null) Pokemon("Name") else Gson().fromJson(value, Pokemon::class.java)

    @JvmStatic
    @TypeConverter
    fun pokemonToString(pokemon: Pokemon?): String = Gson().toJson(pokemon)
}
